panic: duplicate field

## tl;dr

```
$ super -c "SELECT 1 AS x, 2 AS x"
panic: duplicate field: "x"
goroutine 1 [running]:
runtime/debug.Stack()
	/usr/local/opt/go/libexec/src/runtime/debug/stack.go:26 +0x5e
github.com/brimdata/super/runtime/sam/op.(*Catcher).Pull.func1()
	/Users/phil/work/super/runtime/sam/op/catcher.go:25 +0x3d
panic({0x4075d00?, 0xc000468000?})
	/usr/local/opt/go/libexec/src/runtime/panic.go:792 +0x132
github.com/brimdata/super.(*Context).MustLookupTypeRecord(...)
	/Users/phil/work/super/context.go:179
github.com/brimdata/super/runtime/sam/expr.(*recordExpr).Eval(0xc000437c70, {0x4c480998, 0xc00044e000}, {{0x4346540?, 0x54c93e0?}, 0x0?, 0x20?})
	/Users/phil/work/super/runtime/sam/expr/values.go:60 +0x3b3
github.com/brimdata/super/runtime/sam/op/yield.(*Op).Pull(0xc00044f900, 0x0)
	/Users/phil/work/super/runtime/sam/op/yield/yield.go:34 +0x42b
github.com/brimdata/super/runtime/sam/op.(*Single).Pull(0xc00046bec0, 0x5?)
	/Users/phil/work/super/runtime/sam/op/mux.go:120 +0x33
github.com/brimdata/super/runtime/sam/op.(*Catcher).Pull(0xc0c80063846487b9?, 0x39?)
	/Users/phil/work/super/runtime/sam/op/catcher.go:28 +0x5c
github.com/brimdata/super/runtime/exec.(*Query).Pull(0x23960df?, 0x0?)
	/Users/phil/work/super/runtime/exec/query.go:49 +0x3c
github.com/brimdata/super/zbuf.CopyMux(0xc000133ca8, {0x433d580, 0xc00046bef0})
	/Users/phil/work/super/zbuf/mux.go:40 +0x38
github.com/brimdata/super/cmd/super/root.(*Command).Run(0xc0003ae600, {0xc00003c710, 0x0, 0x0})
	/Users/phil/work/super/cmd/super/root/command.go:167 +0x925
github.com/brimdata/super/pkg/charm.path.run({0xc000080c20, 0x1, 0x1}, {0xc00003c710, 0x0, 0x0?})
	/Users/phil/work/super/pkg/charm/path.go:11 +0x7b
github.com/brimdata/super/pkg/charm.(*Spec).Exec(0x548e200, {0xc00003c700, 0x2, 0x2})
	/Users/phil/work/super/pkg/charm/charm.go:71 +0x105
main.main()
	/Users/phil/work/super/cmd/super/main.go:40 +0x5b
```

## Details

Repro is with super commit 9f87359.

As the stack dump indicates, this panic is due to SuperDB not yet supporting duplicate column names. It works fine with unique column names of course.

```
$ super -version
Version: v1.18.0-450-g9f873597e

$ super -c "SELECT 1 AS x, 2 AS y"
{x:1,y:2}
```

If we lack full support for duplicate column names in the first GA SuperDB release, we may want to have a brief error message returned here rather than a panic.

--- a/compiler/semantic/sql.go
+++ b/compiler/semantic/sql.go
@@ -543,8 +543,13 @@ func (a *analyzer) semProjection(sch *selectSchema, args []ast.AsExpr, funcs *ag
 		col := a.semAs(sch, as, funcs)
 		//XXX check for conflict for now, but we're getting rid of this soon
 		if _, ok := conflict[col.name]; ok {
-			a.error(as.ID, fmt.Errorf("%q: conflicting name in projection; try an AS clause", col.name))
+			n := ast.Node(as.Expr)
+			if as.ID != nil {
+				n = as.ID
+			}
+			a.error(n, fmt.Errorf("%q: conflicting name in projection; try an AS clause", col.name))
 		}
+		conflict[col.name] = struct{}{}
 		proj = append(proj, *col)
 		out.columns = append(out.columns, col.name)
 	}

--- a/compiler/ztests/sql/duplicate-as.yaml
+++ b/compiler/ztests/sql/duplicate-as.yaml
@@ -1,0 +1,9 @@
+script: |
+  ! super -c 'SELECT 1 as foo, 2 as foo'
+
+outputs:
+  - name: stderr
+    data: |
+      "foo": conflicting name in projection; try an AS clause at line 1, column 23:
+      SELECT 1 as foo, 2 as foo
+                            ~~~

